src/vkscript/parser_test.cc

```diff
@@ -312,17 +312,19 @@ TEST_F(VkScriptParserTest, VertexDataHeaderFormatString) {
   ASSERT_TRUE(nodes[0]->IsVertexData());
 
   auto* data = nodes[0]->AsVertexData();
-  EXPECT_TRUE(data->GetRows().empty());
 
-  auto& headers = data->GetHeaders();
+  ASSERT_EQ(2U, data->SegmentCount());
 
-  ASSERT_EQ(2U, headers.size());
-  EXPECT_EQ(static_cast<size_t>(0U), headers[0].location);
-  EXPECT_EQ(FormatType::kR32G32_SFLOAT, headers[0].format->GetFormatType());
+  auto& header_0 = data->GetHeader(0);
+  EXPECT_EQ(static_cast<size_t>(0U), header_0.location);
+  EXPECT_EQ(FormatType::kR32G32_SFLOAT, header_0.format->GetFormatType());
+  EXPECT_TRUE(data->GetSegment(0).empty());
 
-  EXPECT_EQ(1U, headers[1].location);
+  auto& header_1 = data->GetHeader(1);
+  EXPECT_EQ(1U, header_1.location);
   EXPECT_EQ(FormatType::kA8B8G8R8_UNORM_PACK32,
-            headers[1].format->GetFormatType());
+            header_1.format->GetFormatType());
+  EXPECT_TRUE(data->GetSegment(1).empty());
 }
 
 TEST_F(VkScriptParserTest, VertexDataHeaderGlslString) {
```

Comment:
Not for this CL, but I'm confused by the  1/int/vec3 attribute header declaration. Is it sane to have a mismatch between the gl_type and the component of the glsl_type ?

Reply:
I have no idea, this is directly from VkRunner, so I have to replicate what they do to maintain compatibility.